packages/core/src/forms/forms.stories.ts

```diff
@@ -1723,125 +1733,130 @@ export const multiColumn = () => {
 export const checkoutForm = () => {
   return html`
     <cds-form-group layout="vertical" cds-layout="container:sm">
-      <p cds-text="section">Billing Address</p>
-
-      <div cds-layout="grid cols@xs:6 gap:lg">
-        <cds-input>
-          <label>first name</label>
-          <input type="text" />
-        </cds-input>
-
-        <cds-input>
-          <label>last name</label>
-          <input type="text" />
-        </cds-input>
-      </div>
-
-      <cds-input>
-        <label>email</label>
-        <input type="email" placeholder="you@example.com" />
-      </cds-input>
-
-      <div cds-layout="grid cols@xs:6 gap:lg">
-        <cds-input>
-          <label>address</label>
-          <input type="text" placeholder="1234 Main St." />
-        </cds-input>
-
-        <cds-input>
-          <label>address 2</label>
-          <input type="text" placeholder="Apartment or Suite" />
-          <cds-control-message>(optional)</cds-control-message>
-        </cds-input>
-      </div>
-
-      <div cds-layout="grid gap:lg">
-        <cds-select cds-layout="col@xs:6 col@sm:5">
-          <label>Country</label>
-          <select>
-            <option>choose...</option>
-            <option>United States</option>
-          </select>
-        </cds-select>
-
-        <cds-select cds-layout="col@xs:6 col@sm:4">
-          <label>State</label>
-          <select>
-            <option>choose...</option>
-            <option>California</option>
-          </select>
-        </cds-select>
+      <div cds-layout="vertical gap:xl">
+        <fieldset cds-layout="vertical gap:md">
```

Comment:
Fieldsets and legends don't fully behave in layouts. I'm not 100% sure why. But you can see I took an alternate strategy of defining my smallest vertical whitespace as the "gap" and adding whitespace with our padding tokens.

Reply:
I ran into this when working on forms. Fieldset and legend dont follow standard CSS layout and will break anything that uses flexbox or css grid and also has several cross browser behavior issues. Overall its not used often because it breaks most CSS. The issue applies to almost all browsers currently. I think we should hold off on this change until we find a robust solution otherwise we will have lots of weird/broken layouts in general in clarity regardless of `cds-layout` usage.

Related links:
https://github.com/jensimmons/cssremedy/issues/50 
https://stackoverflow.com/questions/28078681/why-cant-fieldset-be-flex-containers
https://github.com/philipwalton/flexbugs#flexbug-9 
https://bugs.chromium.org/p/chromium/issues/detail?id=375693
https://bugzilla.mozilla.org/show_bug.cgi?id=984869 

Some options, we rely on role group and aria-labeledby which is what some of the links above suggest. Another option is we create a some kind of wrapper component like `<cds-fieldset>`. Or possibly maybe the most robust solution is to have the current `cds-form-group` create an inner `<fieldset>` if a `<legend>` element is slotted. Then its encapsulated and wont break regular css layouts. Form group is a connivence component to make it easy to layout a group of controls. There can be more than one in a form so it does already closely relate to a fieldset.

```html
<cds-form-group>
  <legend>...</legend>
  <cds-input>...</cds-input>
  <cds-input>...</cds-input>
</cds-form-group>
```

Reply:
Due to the nature and time constraints of this effort, I am inclined to use groups and aria-labelledby. Creating another wrapper component or an inner encapsulation for field sets would take a lot longer. And I honestly don't know if that would fix the issues I've seen.

Reply:
I'd vote to use groups and aria-labeledby as well for now. That way we dont push something out that would need to be rolled back or cause unexpected behavior like legend/fieldset do.

Something like this I would think would be a reasonable solution until we better figure out how to handle fieldset

```html
    <div cds-layout="vertical gap:lg container:sm">
      <cds-form-group role="group" aria-labelledby="billing-address-group" layout="vertical">
        <p id="billing-address-group" cds-text="section">Billing Address</p>
        ...
      </cds-form-group>

      <cds-form-group role="group" aria-labelledby="payment-group" layout="vertical">
        <p id="payment-group" cds-text="section">Payment</p>
        ...
      </cds-form-group>
    </div>
```